mesonbuild/linkers/linkers.py

```diff
@@ -700,14 +700,31 @@ def build_rpath_args(self, env: 'Environment', build_dir: str, from_dir: str,
         return (args, rpath_dirs_to_remove)
 
     def get_win_subsystem_args(self, value: str) -> T.List[str]:
-        if 'windows' in value:
-            args = ['--subsystem,windows']
-        elif 'console' in value:
-            args = ['--subsystem,console']
-        else:
-            raise MesonException(f'Only "windows" and "console" are supported for win_subsystem with MinGW, not "{value}".')
+        # Subsystem affects a single entry at the beginning of PE executables.
+        # It is used to verify that an executable is of the expected
+        # application type before attempting to execute it.
+        subsystems = {
+            "native": "1",
+            "windows": "2",
+            "console": "3",
```

Comment:
Since windows/console are supported without using integers, IMHO we should return the english name (resulting in the same value we used before this PR).